pkg/ingester/transfer.go

```diff
@@ -351,28 +351,36 @@ func (i *Ingester) TransferTSDB(stream client.Ingester_TransferTSDBServer) error
 	return nil
 }
 
-// The passed wireChunks slice is for re-use.
-func toWireChunks(descs []*desc, wireChunks []client.Chunk) ([]client.Chunk, error) {
+// The passed wireChunks and bufs slices are for re-use.
+func toWireChunks(descs []*desc, wireChunks []client.Chunk, bufs []bytes.Buffer) ([]client.Chunk, []bytes.Buffer, error) {
 	if cap(wireChunks) < len(descs) {
 		wireChunks = make([]client.Chunk, 0, len(descs))
 	}
+	if len(bufs) < len(descs) {
```

Comment:
We can be consistent by having only one of `len` or `cap` to check `bufs` and `wireChunks` above.

Reply:
No, they are used in different ways.

Reply:
Fair enough